pyteal/ast/scratch.py

```diff
@@ -59,14 +59,12 @@ def load(self, type: TealType = TealType.anytype) -> "ScratchLoad":
         """
         return ScratchLoad(self, type)
 
+    def __repr__(self):
+        return "ScratchSlot({})".format(self.id)
+
     def __str__(self):
         return "slot#{}".format(self.id)
 
-    def __eq__(self, other):
```

Comment:
I had to remove this so that `ScratchSlot(1) != ScratchSlot(1)`, which is required in order to detect duplicate requested slot IDs when I add all slots to a set.